src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -215,16 +215,10 @@ private void WriteMvidTo(ModuleMetadata metadata, ObjectWriter writer, Cancellat
             PortableExecutableReference reference, ObjectWriter writer, CancellationToken cancellationToken)
         {
             WritePortableExecutableReferenceHeaderTo(reference, SerializationKinds.Bits, writer, cancellationToken);
-            WritePortableExecutableReferenceMetadataTo(reference, writer, cancellationToken);
 
-            // TODO: what I should do with documentation provider? it is not exposed outside
-        }
+            WriteTo(reference.GetMetadata(), writer, cancellationToken);
 
-        private void WritePortableExecutableReferenceMetadataTo(
-            PortableExecutableReference reference, ObjectWriter writer, CancellationToken cancellationToken)
-        {
-            var metadata = GetMetadata(reference);
-            WriteTo(metadata, writer, cancellationToken);
+            // TODO: what I should do with documentation provider? it is not exposed outside
```

Comment:
I have issue with bunch of TODOs in them. I am actively going them one by one. so I don't think for now I need link into the code yet.

things I decide not to fix in short term I will add those link.